source/common/rds/rds_route_config_provider_impl.cc

```diff
@@ -15,13 +15,13 @@ RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl(
     return std::make_shared<ThreadLocalConfig>(initial_config);
   });
   // It should be 1:1 mapping due to shared rds config.
-  ASSERT(!subscription_->routeConfigProvider().has_value());
-  subscription_->routeConfigProvider().emplace(this);
+  ASSERT(subscription_->routeConfigProvider() == nullptr);
+  subscription_->routeConfigProvider() == this;
```

Comment:
Did you mean to use `=` instead of `==` here?

Reply:
Duh, of course, thanks for catching this (before tests did :))